python/semantic_kernel/connectors/ai/google/google_ai/services/google_ai_text_embedding.py

```diff
@@ -0,0 +1,107 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import sys
+from typing import Any
+
+import google.generativeai as genai
+from google.generativeai.types.text_types import BatchEmbeddingDict
+from numpy import array, ndarray
+from pydantic import ValidationError
+
+from semantic_kernel.connectors.ai.embeddings.embedding_generator_base import EmbeddingGeneratorBase
+from semantic_kernel.connectors.ai.google.google_ai.google_ai_prompt_execution_settings import (
+    GoogleAIEmbeddingPromptExecutionSettings,
+)
+from semantic_kernel.connectors.ai.google.google_ai.google_ai_settings import GoogleAISettings
+from semantic_kernel.connectors.ai.google.google_ai.services.google_ai_base import GoogleAIBase
+from semantic_kernel.connectors.ai.prompt_execution_settings import PromptExecutionSettings
+from semantic_kernel.exceptions.service_exceptions import ServiceInitializationError
+
+if sys.version_info >= (3, 12):
+    from typing import override  # pragma: no cover
+else:
+    from typing_extensions import override  # pragma: no cover
+
+
+class GoogleAITextEmbedding(GoogleAIBase, EmbeddingGeneratorBase):
+    """Google AI Text Embedding Service."""
+
+    def __init__(
+        self,
+        ai_model_id: str | None = None,
+        api_key: str | None = None,
+        service_id: str | None = None,
+        env_file_path: str | None = None,
+        env_file_encoding: str | None = None,
+    ) -> None:
+        """Initialize the Google AI Text Embedding service.
+
+        If no arguments are provided, the service will attempt to load the settings from the environment.
+        The following environment variables are used:
+        - GOOGLE_AI_AI_MODEL_ID
+        - GOOGLE_AI_API_KEY
+
+        Args:
+            ai_model_id (str | None): The AI model ID. (Optional)
+            api_key (str | None): The API key. (Optional)
+            service_id (str | None): The service ID. (Optional)
+            env_file_path (str | None): The path to the .env file. (Optional)
+            env_file_encoding (str | None): The encoding of the .env file. (Optional)
+
+        Raises:
+            ServiceInitializationError: If an error occurs during initialization.
+        """
+        try:
+            google_ai_settings = GoogleAISettings.create(
+                ai_model_id=ai_model_id,
```

Comment:
Last comment :) this is `embedding_model_id` in other connectors.

Reply:
Yeah, I think we can do this.